examples/document-tokenizer.rs

```diff
@@ -221,9 +241,33 @@ fn annotate(node : Node, root: &Node, range: &DNMRange, dnm: &DNM, dom: &DOM) ->
         act_end.unlink();
         node.add_child(&act_end).unwrap();
     }
+
     return true;
 }
 
+/* Takes a possibly invalid byte index to plaintext and returns a close and valid byte index. A byte index is invalid iff it does
+   not point to a char boundary */
+fn fix_byte_index(mut byte_index : usize, plaintext : String) -> usize {
```

Comment:
I think this is the core fix needed for the DNM ranges / tokenization boundaries - they must always point at a char boundary. An approach I may borrow and provide fixes natively in the methods affected.